starport-core/src/main/scala/com/krux/starport/StartScheduledPipelines.scala

```diff
@@ -218,35 +137,12 @@ object StartScheduledPipelines extends StarportActivity {
         new ForkJoinPool(parallel * Runtime.getRuntime.availableProcessors)
       )
 
-    val successfulPipelines = metrics.register("counter.successful-pipeline-deployment-count", new Counter())
-    val failedPipelines = metrics.register("counter.failed-pipeline-deployment-count", new Counter())
-
-    parPipelineModels.foreach { p =>
-
-      val timerInst = scheduleTimer.time()
-
-      logger.info(s"deploying pipleine ${p.name}")
+    parPipelineModels.foreach(p => processPipeline(taskDispatcher, p, options, localJars(p.jar), dispatchedPipelines, failedPipelines))
```

Comment:
we also make sure taskDispatcher is thread safe, since it will be passed to different threads for execution. Can you double check?

Reply:
yes, I need to make the mutable Queue access threadsafe. Thank you for pointing this out.